biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts

```diff
@@ -1,66 +1,185 @@
-import * as jsgraphs from 'js-graph-algorithms';
+import * as graphlib from "graphlib";
 import {
   StateGraphVertex,
   StateGraph,
   StateGraphEdge,
+  Distance,
+  ItemPosition,
 } from '@biogrid/grid-simulator';
+import { GridItem } from 'libs/grid-simulator/src/lib/grid-item';
+import { GRID_ITEM_NAMES, ShortestDistances } from '../config';
 
 export class BiogridState implements StateGraph {
-  // Stores the wires which connect each grid item to each other and the distance between them
-  graph: jsgraphs.WeightedDiGraph;
-  // Because the graph stores numbers as vertices, graphIndexToVertex is needed to
-  // convert between a vertex in the graph and the datastructure it represents
-  graphIndexToVertex: StateGraphVertex[];
+  private graph: graphlib.Graph;
 
+  // TODO think about implement it StateGraphVertex[] as an object of key: name -> value: StateGraphVertex
   constructor(vertices: StateGraphVertex[]) {
-    this.graph = new jsgraphs.WeightedDiGraph(vertices.length);
-    this.graphIndexToVertex = new Array<StateGraphVertex>(vertices.length);
-    vertices.forEach((vertex, i) => {
-      this.graphIndexToVertex[i] = vertex;
-    });
+    // Directed so as to have two edges between A and B, but in opposite directions
+    this.graph = new graphlib.Graph({directed: true});
+
+    // Initialize the graph with a grid which is a gridItem and has position (0, 0) to keep track of where the items are placed on the map
+    const grid: GridItem = {
+      name: GRID_ITEM_NAMES.GRID,
+      getRelativePosition() {
+        return {x: 0, y: 0};
+      }
+    }
+    this.graph.setNode(grid.name, (grid as GridItem));
+
+    // Add all the vertices as nodes/vertices of the graph, with a name for
+    // the particular grid item and label which is data for the particular vertex as the GridItem itself
+    vertices.map(vertex => this.graph.setNode(vertex.name, (vertex as GridItem)));
+
+
+    // Add all the edges that can be formed into the graph, read the add method for how it is done
+    vertices.map(vertex => this.connectNewVertex(vertex));
   }
 
+  /**
+   * Method to return the graph for the states of the grid
+   */
   public getGraph() {
     return this.graph;
   }
 
-  public convertStateGraphToMST() {
-    this.graphIndexToVertex.forEach((vertex, i) => {
-      this.addCompletelyConnectedVertex(i, vertex);
-    });
-    this.graph = this.getMinimumSpanningTree( this.graph);
+  /**
+   * cloneStateGraph is used to clone the graph for use in the brain.
+   */
+  public cloneStateGraph(): graphlib.Graph {
+    return graphlib.json.read(graphlib.json.write(this.graph));
+  }
+
+  /**
+   * getJsonGraph returns the json details of the graph
+   */
+  public getJsonGraph() {
+    return graphlib.json.write(this.graph);
   }
 
-  public getGridItem(ind: number) {
-    return this.graphIndexToVertex[ind];
+  /**
+   * setnewStateGraph
```

Comment:
update comment or remove

Reply:
I will update the comments, thanks for highlighting this out